File: clojure/test/dayNN_test.clj
(ns dayNN-test
  (:require [clojure.test :refer [deftest is testing use-fixtures]]
            dayNN))

;; https://stuartsierra.com/2016/05/19/fixtures-as-caches
(def ^:dynamic data)
(defn load-data [f]
  (binding [data (slurp "/data/dayNN.txt")] 
    (f)))
(use-fixtures :each load-data)

(def EXAMPLE1 (clojure.string/trim "
"))

(deftest dayNN-examples-test
  (is (and
    (= nil (dayNN/part1 nil))
    (= nil (dayNN/part2 nil)))))

(deftest dayNN-part1-test
  (is (= nil (dayNN/part1 nil))))

(deftest dayNN-part2-test
  (is (= nil (dayNN/part2 nil))))

File: elixir/test/dayNN_test.exs
defmodule DayNNTest do
  use ExUnit.Case, async: true
  import Advent.DayNN

  @moduletag :dayNN
  @example1 """
  """

  setup_all do
    {:ok, data} = File.read("/data/dayNN.txt")
    {:ok, data: data}
  end

  describe "examples" do
    @tag examples: :part1
    test "part1" do
      assert part1(@example1) == nil
    end

    @tag examples: :part2
    test "part2" do
      assert part2(@example1) == nil
    end
  end

  describe "solutions" do
    @tag solutions: :part1
    test "part1", state do
      assert part1(state[:data]) == nil
    end

    @tag solutions: :part2
    test "part2", state do
      assert part2(state[:data]) == nil
    end
  end
end
